models/classes/ItemPreviewer.php

```diff
@@ -0,0 +1,279 @@
+<?php
+
+namespace oat\taoQtiTest\models;
+
+use oat\taoDelivery\model\RuntimeService;
+use oat\taoItems\model\pack\ItemPack;
+use oat\taoItems\model\pack\Packer;
+use oat\taoQtiItem\model\qti\Service;
+use oat\taoQtiItem\model\QtiJsonItemCompiler;
+use oat\taoQtiTest\models\container\QtiTestDeliveryContainer;
+use oat\taoResultServer\models\classes\ResultServerService;
+use Zend\ServiceManager\ServiceLocatorAwareInterface;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+
+class ItemPreviewer implements ServiceLocatorAwareInterface
+{
+    use ServiceLocatorAwareTrait;
+
+    /**
+     * @var string
+     */
+    private $resultId;
+
+    /**
+     * @var string
+     */
+    private $itemDefinition;
+
+    /**
+     * @var \core_kernel_classes_Resource
+     */
+    private $delivery;
+
+    /**
+     * @var \tao_models_classes_service_FileStorage
+     */
+    private $fileStorage;
+
+    /**
+     * @var string
+     */
+    private $itemUri;
+
+    /**
+     * @var \tao_models_classes_service_StorageDirectory
+     */
+    private $itemPublicDir;
+
+    /**
+     * @var \tao_models_classes_service_StorageDirectory
+     */
+    private $itemPrivateDir;
+
+    /**
+     * @var string
+     */
+    private $userDataLang;
+
+    /**
+     * @var array
+     */
+    private $itemHrefs = [];
+
+    public function __construct()
+    {
+        $this->fileStorage = \tao_models_classes_service_FileStorage::singleton();
+    }
+
+    /**
+     * @param string $resultId
+     * @return ItemPreviewer
+     */
+    public function setResultId($resultId)
```

Comment:
Should we set the results ? If the result id is used to retrieve the test taker and then the language, in this extension we should better set the language (which is computed at the place this is called/instantiated)

Reply:
Maybe, that can be a option but in that case the code of language computing will have to be moved into the controller which would make it more "crammed".